src/pipelinedefinition.cpp

```diff
@@ -31,10 +34,51 @@ Status toNodeKind(const std::string& str, NodeKind& nodeKind) {
     return StatusCode::PIPELINE_NODE_WRONG_KIND_CONFIGURATION;
 }
 
+Status PipelineDefinition::validate(ModelManager& manager) {
+    struct ValidationResultNotifier {
+        ValidationResultNotifier() {}
+        ~ValidationResultNotifier() {
+            if (passed) {
+                // status.notifyValidationPassed();
```

Comment:
This class has no information about pipeline therefore no info about its status. How are you planning to use it exactly in next MR?

Reply:
Either I need to expose notifyValidationPassed in public interface of PipelineDefinition
or make this class friend to class PipelineDefinition.